optimize/backend/src/main/java/io/camunda/optimize/service/db/repository/es/TaskRepositoryES.java

```diff
@@ -130,6 +135,25 @@ public boolean tryDeleteByQueryRequest(
                     .waitForCompletion(false)
                     .conflicts(Conflicts.Proceed));
 
+    if (configuration.getConnection().isHealthCheckEnabled()) {
+      return async(query, deletedItemIdentifier, request);
+    } else {
+      return sync(request);
+    }
+  }
+
+  private boolean sync(final DeleteByQueryRequest request) {
+    try {
+      final Long deleted = esClient.submitDeleteTask(request).deleted();
+      return deleted != null && deleted > 0L;
+    } catch (final IOException e) {
+      throw new OptimizeRuntimeException(
+          "Error while trying to read Elasticsearch task status with ID");
+    }
+  }
+
+  private boolean async(
```

Comment:
do we also need corresponding logic for OS?

Reply:
After discussion with @RomanJRW we summarize that project does not need solution for opensearch part, thats why we will go with implementation of solution for elasticsearch.